### PR TITLE
Add initial canonical layout implementations

### PR DESCRIPTION
**Pull request by alexvanyo**
Adds the initial implementations of canonical layouts using the current set of layout technologies.

There are 8 in total:

- List-Detail
  - with activity embedding
  - with Compose (`TwoPane`)
  - with views (`SlidingPaneLayout`)
- Feed
  - with Compose (`LazyVerticalGrid`)
  - with views (`RecyclerView`)
- Supporting Panel
   - with Compose (`TwoPane`)
   - with fragments
   - with views
   
These samples are barebones, just to show the layout behavior and standalone examples of using each of these technologies to achieve implementations of the canonical layouts.

### CODE REVIEW

File: CanonicalLayouts/feed-compose/app/src/main/java/com/example/feedcompose/ui/components/TopAppBar.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import com.example.feedcompose.R
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+internal fun TopAppBar(onBackPressed: () -> Unit = {}) {
+    SmallTopAppBar(
```
Comment (Gurupreet) [inline CanonicalLayouts/feed-compose/app/src/main/java/com/example/feedcompose/ui/components/TopAppBar.kt:32]: New Material Compose is available to update which has renamed this to just TopAppBar.

File: CanonicalLayouts/feed-compose/app/src/main/java/com/example/feedcompose/ui/components/TopAppBar.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        navigationIcon = { BackButton(onBackPressed) }
+    )
+}
+
+@Composable
+private fun BackButton(onBackPressed: () -> Unit) {
+    IconButton(onClick = onBackPressed) {
+        Icon(
+            painter = painterResource(id = R.drawable.ic_baseline_arrow_back_24),
+            contentDescription = null
```
Comment (Gurupreet) [inline CanonicalLayouts/feed-compose/app/src/main/java/com/example/feedcompose/ui/components/TopAppBar.kt:43]: I know it's sample but we can add values to these to keep accessibility focused.

File: CanonicalLayouts/feed-compose/app/src/main/java/com/example/feedcompose/ui/theme/Color.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.feedcompose.ui.theme
+
+import androidx.compose.ui.graphics.Color
+
+val Purple80 = Color(0xFFD0BCFF)
```
Comment (Gurupreet) [inline CanonicalLayouts/feed-compose/app/src/main/java/com/example/feedcompose/ui/theme/Color.kt:21]: Nit: You can generate full theme easily using Material theme builder: https://m3.material.io/theme-builder#/custom

**Review by Gurupreet — APPROVED** [review]
Reviewed most of the compose code. 
It looks good considering it's early setup. Left few non blocking comments.

**Comment by alexvanyo** [comment]
Merging to be able to have public links to these, I'm going to create a follow-up PR with the README and @Gurupreet 's initial feedback!

**Pull request merged**